<launch>
    <arg name="fixed_frame" default="world"/>

    <param name="robot_description" textfile="$(find hybrid_a_star_zm0612)/model/car_model/default.urdf"/>
    <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher"/>
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher"/>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find hybrid_a_star_zm0612)/rviz/display_hybrid_astar.rviz"/>

    <!--static_transform_publisher x y z yaw pitch roll frame_id child_frame_id period_in_ms-->
    <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster"
          args="0 0 0 0 0 0 world map 100"/>

    <node pkg="map_server" type="map_server" name="map_server" args="$(find hybrid_a_star_zm0612)/maps/map.yaml"/>

    <node pkg="hybrid_a_star_zm0612" type="hybrid_a_star_zm0612" name="hybrid_a_star_zm0612" output="screen">
        
        <param name="planner/steering_angle" value="30.0"/><!-- <最大转弯角度的值，然后将这个值离散化为num份。向左和向右分开的> -->
        <param name="planner/steering_angle_discrete_num" value="2"/>

        <param name="planner/wheel_base" value="0.8"/><!-- <车前后两个轮的距离，用于前向采样积分，运动线模型 R = wheel_base/theta> -->
        <param name="planner/segment_length" value="1.0"/><!-- <每次采样的路径长度> -->
        <param name="planner/segment_length_discrete_num" value="8"/><!-- <采样的这一段线，由几个点表示> -->

        <param name="planner/steering_penalty" value="1.2"/>
        <param name="planner/steering_change_penalty" value="1.5"/>
        <param name="planner/reversing_penalty" value="1.5"/>
        <param name="planner/shot_distance" value="15.0"/>

        <param name="planner/reverse_enable" value="1"/> <!--是否允许倒车-->

        <param name="planner/vehicle_length" value="3.0"/>
        <param name="planner/vehicle_width" value="1.5"/>
        <param name="planner/vehicle_rear_dis" value="0.5"/>

    </node>

</launch>
